### DISCUSSION TITLE
Support for Deferred Deep Linking

### DISCUSSION THREAD

**Discussion by Muddassir-Farooq-official** [flet-dev/flet#4135] [Ideas]
### Duplicate Check

- [X] I have searched the [opened issues](https://github.com/flet-dev/flet/issues) and there are no duplicates

### Describe the requested feature

**Feature Description:**

I would like to request the addition of Deferred Deep Linking support in Flet applications. Deferred deep linking allows an app to direct users to specific in-app content even if the app was not previously installed. After the user installs the app, they should be able to continue to the desired content seamlessly.

### Suggest a solution

**Why This Feature is Valuable:**

- **Enhanced User Experience**: Users would be directed to the exact content they intended to access after installing the app. This is particularly useful in marketing campaigns, where users are directed to promotions, events, or product details.

- **Engagement & Retention**: Providing users with a more seamless post-installation experience would likely improve engagement and retention rates. Users don't have to manually navigate through the app to find what they are looking for.
-
- **Common in Modern Apps**: Deferred deep linking is a standard feature in many mobile and web applications. Having this feature in Flet would make it easier to build competitive, user-friendly apps with this functionality.

### Additional details

**Suggested Implementation:**

Here’s how the feature might work:
1. Flet can integrate with services like Firebase Dynamic Links, which supports deferred deep linking.
2. A URL handler in Flet could listen for incoming deep links and route users to specific sections of the app based on the link.
3. The system could remember the link post-installation and ensure that the user is directed to the correct page after the app is installed.

### **Benefits for Flet Developers:**
- Developers can use deep links for targeted marketing, improving the app's discoverability and user experience.
- Makes Flet apps more versatile and competitive with other frameworks that already support deep linking.

---

**Example Use Case**:
A marketing campaign sends out a link for users to download a Flet app. If users click the link but don’t have the app installed, they are redirected to the app store. After installing the app, they are taken directly to the campaign's page or special promotion, improving conversion rates.

---

I hope the community and the Flet team consider this request, as it will significantly enhance the functionality and user experience of Flet-based applications

Implement this also

<intent-filter>
    <action android:name="android.intent.action.VIEW" />
    <category android:name="android.intent.category.DEFAULT" />
    <category android:name="android.intent.category.BROWSABLE" />
    <data android:scheme="https" android:host="www.yourdomain.com" />
</intent-filter>

**Comment by FeodorFitsner** [comment]
Could you provide some more info about how this could be implemented in Flutter?

**Comment by Muddassir-Farooq-official** [comment]
> Could you provide some more info about how this could be implemented in Flutter?

This is Flutter Official Tutorial On Deep Linking

https://youtu.be/KNAb2XL7k2g?si=blT_IVSrU8i5gI6G

**Comment by Muddassir-Farooq-official** [comment]
> > Could you provide some more info about how this could be implemented in Flutter?
> 
> This is Flutter Official Tutorial On Deep Linking
> 
> https://youtu.be/KNAb2XL7k2g?si=blT_IVSrU8i5gI6G

It's a request kindly approve project.toml project and merge mobile ads

**Comment by FeodorFitsner** [comment]
That video is about deep linking which is now available in [Flet pre-release](https://flet.dev/blog/flet-new-packaging-pre-release).

I did some googling about deferred deep linking and didn't find anything useful. Let switch to discussions.

**Reply by Muddassir-Farooq-official** [reply]
@FeodorFitsner is it possible to add functionality in flet to read and display pdf files and csv kindly launch in this new pre release it's too important
